x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/job_map.tsx

```diff
@@ -37,102 +37,78 @@ ${theme.euiColorLightShade}`,
-  const [error, setError] = useState(undefined);
+  const [itemsDeleted, setItemsDeleted] = useState<boolean>(false);
+  const [resetCy, setResetCy] = useState<boolean>(false);
+  const {
+    elements,
+    error,
+    getDataWrapper,
+    isLoading,
+    message,
+    nodeDetails,
+    setElements,
+    setError,
+  } = useFetchAnalyticsMapData();
 
   const {
-    services: { notifications },
+    services: {
+      notifications,
+      application: { navigateToUrl },
+    },
   } = useMlKibana();
+  const urlGenerator = useMlUrlGenerator();
 
-  const getDataWrapper = async (params?: GetDataObjectParameter) => {
-    const { id, type } = params ?? {};
-    const treatAsRoot = id !== undefined;
-    let idToUse: string;
-
-    if (id !== undefined) {
-      idToUse = id;
-    } else if (modelId !== undefined) {
-      idToUse = modelId;
-    } else {
-      idToUse = analyticsId as string;
-    }
-
-    await getData(
-      idToUse,
-      treatAsRoot,
-      modelId !== undefined && treatAsRoot === false ? JOB_MAP_NODE_TYPES.TRAINED_MODEL : type
-    );
+  const redirectToAnalyticsManagementPage = async () => {
+    const url = await urlGenerator.createUrl({ page: ML_PAGES.DATA_FRAME_ANALYTICS_JOBS_MANAGE });
+    await navigateToUrl(url);
   };
 
-  const getData = async (idToUse: string, treatAsRoot: boolean, type?: string) => {
-    // Pass in treatAsRoot flag - endpoint will take job or index to grab jobs created from it
-    // TODO: update analyticsMap return type here
-    const analyticsMap: any = await ml.dataFrameAnalytics.getDataFrameAnalyticsMap(
-      idToUse,
-      treatAsRoot,
-      type
-    );
-
-    const { elements: nodeElements, details, error: fetchError } = analyticsMap;
-
-    if (fetchError !== null) {
-      setError(fetchError);
-    }
-
-    if (nodeElements && nodeElements.length === 0) {
-      notifications.toasts.add(
-        i18n.translate('xpack.ml.dataframe.analyticsMap.emptyResponseMessage', {
-          defaultMessage: 'No related analytics jobs found for {id}.',
-          values: { id: idToUse },
-        })
-      );
-    }
+  if (message !== undefined) {
+    notifications.toasts.add(message);
+  }
 
-    if (nodeElements && nodeElements.length > 0) {
-      if (treatAsRoot === false) {
-        setElements(nodeElements);
-        setNodeDetails(details);
-      } else {
-        const uniqueElements = uniqWith([...nodeElements, ...elements], isEqual);
-        setElements(uniqueElements);
-        setNodeDetails({ ...details, ...nodeDetails });
-      }
+  const updateElements = (nodeId: string, nodeLabel: string, destIndexNode?: string) => {
+    // If removing the root job just go back to the jobs list
+    if (nodeLabel === analyticsId) {
+      redirectToAnalyticsManagementPage();
+    } else {
+      // Remove job element
+      const filteredElements = elements.filter((e: any) => {
+        // Filter out job node and related edges, including trained model node.
+        let condition =
```

Comment:
`condition` doesn't seem descriptive enough for this variable. maybe `isNodeIdElement` or something similar

Reply:
Updated to a more descriptive name in 626ab62bb02d68546b25cf3cbf527a0fd88cc1a9